sklearn/model_selection/_classification_threshold.py

```diff
@@ -640,7 +641,18 @@ class TunedThresholdClassifierCV(BaseThresholdClassifier):
 
         * a string associated to a scoring function for binary classification
           (see :ref:`scoring_parameter`);
-        * a scorer callable object created with :func:`~sklearn.metrics.make_scorer`;
+        * a scorer callable object created with :func:`~sklearn.metrics.make_scorer`.
+
+        Note that scoring objective should introduce a trade-off between false
+        negatives and false positives, otherwise the tuned threshold would be
+        trivial and the resulting classifier would be equivalent to constantly
+        classifiying one of the two possible classes. This would be the case
+        when passing scoring="precision" or scoring="recall" for instance.
+        Furthermore, the scoring objective should evaluate thresholded
+        classifier predictions: as a result, metrics such as ROC AUC, Average
+        Precision, log loss or the Brier score are not valid scoring metrics in
+        this context since they are all designed to evaluate unthresholded
+        class-membership confidence scores.
```

Comment:
Could this be made much shorter?

Reply:
I am not sure what to remove besides maybe cutting the last sentence:

https://github.com/scikit-learn/scikit-learn/pull/29082/files#r1616901942